runtime/connectors/s3/s3.go

```diff
@@ -192,3 +181,34 @@ func getAwsSessionConfig(ctx context.Context, conf *Config, bucket string) (*ses
 
 	return sess, nil
 }
+
+func resolvedCredentials(env *connectors.Env) (*credentials.Credentials, error) {
+	providers := make([]credentials.Provider, 0)
+	useHostCred := env.Variables["use_host_credentials"] != "false" // true by default
+	if useHostCred {
+		// The chain used here is a duplicate of defaults.CredProviders(), but without the remote credentials lookup (since they resolve too slowly).
+		providers = append(providers, &credentials.EnvProvider{}, &credentials.SharedCredentialsProvider{Filename: "", Profile: ""})
+	} else {
+		// in case host credential lookup is disabled we need to rely on user provided access keys only
+		staticProvider := &credentials.StaticProvider{}
+		staticProvider.AccessKeyID = env.Variables["aws_access_key_id"]
+		staticProvider.SecretAccessKey = env.Variables["aws_secret_access_key"]
+		staticProvider.SessionToken = env.Variables["aws_session_token"]
+		staticProvider.ProviderName = credentials.StaticProviderName
+		providers = append(providers, staticProvider)
```

Comment:
Is it safe to add this provider when none of these keys are set? Or should we check for the env var to exist before adding?

Reply:
Yes it is safe to do so. In case user doesn't set access key id and secret access key the credentials retrieval will fail and the lookup will proceed to next provider in chain
reference : https://github.com/aws/aws-sdk-go/blob/33b50e7d3b3940d98236cb33ae377d2f12696215/aws/credentials/static_provider.go#L42